adapter/internal/discovery/xds/rate_limiter_cache_test.go

```diff
@@ -782,3 +783,68 @@ func TestAddSubscriptionLevelRateLimitPolicy(t *testing.T) {
 
 	assert.Equal(t, expectedPolicies, rlsPolicyCache.metadataBasedPolicies)
 }
+
+func TestSubscriptionRateLimitPolicyEventDataHandling(t *testing.T) {
+	policy := types.SubscriptionPolicy{
+		Name: "policyToCheckDeletion",
+		DefaultLimit: &types.SubscriptionDefaultLimit{
+			QuotaType: "requestCount",
+			RequestCount: &types.SubscriptionRequestCount{
+				RequestCount: 6100,
+				TimeUnit:     "hour",
+			},
+		},
+		Organization: "org2",
+		RateLimitCount: 61,
+		RateLimitTimeUnit: "min",
+		StopOnQuotaReach: true,
+	}
+	AddSubscriptionLevelRateLimitPolicy(policy)
```

Comment:
Are we calling this method when we get a notification event?

Reply:
yes. We are calling this method.

Reply:
Ack